WarehouseManagement.Core/Services/MarkerService.cs

```diff
@@ -0,0 +1,213 @@
+                    .DeliveriesMarkers.Select(dm => new MarkerDeliveryDto
+                    {
+                        DeliveryId = dm.DeliveryId,
+                        DeliverySystemNumber = dm.Delivery.SystemNumber
+                    })
+                    .ToList(),
+                Vendors = m
+                    .VendorsMarkers.Select(vm => new MarkerVendorDto
+                    {
+                        VendorId = vm.VendorId,
+                        VendorName = vm.Vendor.Name,
+                        VendorSystemNumber = vm.Vendor.SystemNumber
+                    })
+                    .ToList(),
+                Zones = m
+                    .ZonesMarkers.Select(zm => new MarkerZoneDto
+                    {
+                        ZoneId = zm.ZoneId,
+                        ZoneName = zm.Zone.Name
+                    })
+                    .ToList()
+            })
+            .FirstOrDefaultAsync();
+    }
+
+    public async Task RestoreAsync(int id, string userId)
+    {
+        var marker = await repository.GetByIdWithDeletedAsync<Marker>(id);
+
+        if (marker == null)
+        {
+            throw new KeyNotFoundException($"Marker with ID {id} not found.");
+        }
+
+        marker.LastModifiedByUserId = userId; //TODO: Should the user that undeleted be assigned to LastModified user ?
+        repository.UnDelete(marker);
+
+        await repository.SaveChangesWithLogAsync(); //TODO: Should we log the Undeletion process or not ?
+    }
+
+    public async Task<bool> IsDeletedById(int id)
+    {
+        return await repository.AllWithDeleted<Marker>().AnyAsync(m => m.Id == id);
+    }
+
+    public async Task<IEnumerable<MarkerDto>> GetDeletedMarkersAsync()
+    {
+        return await repository
+            .AllWithDeletedReadOnly<Marker>()
+            .Where(m => m.IsDeleted)
+            .Select(m => new MarkerDto
+            {
+                Name = m.Name,
+                Deliveries = m
+                    .DeliveriesMarkers.Select(dm => new MarkerDeliveryDto
+                    {
+                        DeliveryId = dm.DeliveryId,
+                        DeliverySystemNumber = dm.Delivery.SystemNumber
+                    })
+                    .ToList(),
+                Vendors = m
+                    .VendorsMarkers.Select(vm => new MarkerVendorDto
+                    {
+                        VendorId = vm.VendorId,
+                        VendorName = vm.Vendor.Name,
+                        VendorSystemNumber = vm.Vendor.SystemNumber
+                    })
+                    .ToList(),
+                Zones = m
+                    .ZonesMarkers.Select(zm => new MarkerZoneDto
+                    {
+                        ZoneId = zm.ZoneId,
+                        ZoneName = zm.Zone.Name
+                    })
+                    .ToList()
+            })
+            .ToListAsync();
+    }
+
+    public async Task<List<string>> IsMarkerInUseAsync(int markerId)
```

Comment:
why dont you just return a boolean if anything is found? This method is a question - meaning you should expect a yes/no (boolean). If you think at some point we will need to know where a marker is used then rename the function to `GetMarkerUsages` or change this into a boolean and create a new that will get the usages.

Reply:
Yes this is my mistake. First i made it as bool returning method, but later I remembered that maybe it's better to return which tables are using the marker and forgot to change the name.